packages/@aws-cdk/aws-eks/lib/kubectl-layer.ts

```diff
@@ -0,0 +1,75 @@
+import lambda = require('@aws-cdk/aws-lambda');
+import { CfnResource, Construct, Stack, Token } from '@aws-cdk/core';
+import crypto = require('crypto');
+
+export interface KubectlLayerProps {
+  /**
+   * The semantic version of the kubectl AWS Lambda Layer SAR app to use.
+   *
+   * @default '1.13.7'
+   */
+  readonly version?: string;
+}
+
+/**
+ * An AWS Lambda layer that includes kubectl and the AWS CLI.
+ *
+ * @see https://github.com/aws-samples/aws-lambda-layer-kubectl
+ */
+export class KubectlLayer extends Construct implements lambda.ILayerVersion {
+
+  /**
+   * Gets or create a singleton instance of this construct.
+   */
+  public static getOrCreate(scope: Construct, props: KubectlLayerProps = {}): KubectlLayer {
+    const stack = Stack.of(scope);
+    const id = 'kubectl-layer-8C2542BC-BF2B-4DFE-B765-E181FD30A9A0';
+    const exists = stack.node.tryFindChild(id) as KubectlLayer;
+    if (exists) {
+      return exists;
+    }
+
+    return new KubectlLayer(stack, id, props);
+  }
+
+  /**
+   * The ARN of the AWS Lambda layer version.
+   */
+  public readonly layerVersionArn: string;
+
+  /**
+   * All runtimes are compatible.
+   */
+  public readonly compatibleRuntimes?: lambda.Runtime[] = undefined;
+
+  constructor(scope: Construct, id: string, props: KubectlLayerProps = {}) {
+    super(scope, id);
+
+    const uniqueId = crypto.createHash('md5').update(this.node.path).digest("hex");
+    const version = props.version || '1.13.7';
+
+    this.stack.templateOptions.transforms = [ 'AWS::Serverless-2016-10-31' ]; // required for AWS::Serverless
```

Comment:
This needs a check for existing transforms and conflicts therein.

Reply:
Ping. This is still a recipe for disaster if people combine with other transforms, or other versions of the Serverless transform.

Reply:
I'll add some support in `Stack` for this (`stack.addTransform()`). Let me open an issue.